src/posix/platform/radio_url.cpp

```diff
@@ -0,0 +1,224 @@
+    " gpio-reset-line[=line-offset]"                                                                    \
+    "                           The offset index of `R̅E̅S̅` pin for the associated GPIO device.\n"  \
+    " spi-mode[=mode]           Specify the SPI mode to use (0-3).\n"                                   \
+    " spi-speed[=hertz]         Specify the SPI speed in hertz.\n"                                      \
+    " spi-cs-delay[=usec]       Specify the delay after C̅S̅ assertion, in µsec.\n"                  \
+    " spi-reset-delay[=ms]      Specify the delay after R̅E̅S̅E̅T̅ assertion, in milliseconds.\n"  \
+    " spi-align-allowance[=n]   Specify the maximum number of 0xFF bytes to clip from start of\n"       \
+    "                           MISO frame. Max value is 16.\n"                                         \
+    " spi-small-packet=[n]      Specify the smallest packet we can receive in a single transaction.\n"  \
+    "                           (larger packets will require two transactions). Default value is 32.\n"
+#else
+#define RADIO_URL_HELP_BUS                                                                      \
+    "Radio URL:\n"                                                                              \
+    "spinel+hdlc+uart://${PATH_TO_UART_DEVICE}?${arguments} for real uart device\n"             \
+    "spinel+hdlc+fortpty://${PATH_TO_UART_DEVICE}?${arguments} for forking a pty subprocess.\n" \
+    "arguments:\n"                                                                              \
+    " uart-parity[=even|odd]     Uart parity config, optional.\n"                               \
+    " uart-stop[=number-of-bits] Uart stop bit, default is 1.\n"                                \
+    " baudrate[=baudrate]        Uart baud rate, default is 115200.\n"                          \
+    " arg[=argument string]      Command line arguments for subprocess, can be repeated.\n"
+#endif
+    return RADIO_URL_HELP_BUS " no-reset                   Skip resetting the radio device.\n"
+                              " ncp-dataset                Retrieve dataset from ncp.\n"
+                              " max-power-table            Max power table for each channel, splitted by comma.\n";
+}
+
+namespace ot {
+namespace Posix {
+
+Arguments::Arguments(const char *aUrl)
+{
+    char *url = &mUrl[0];
+
+    mPath  = NULL;
+    mStart = NULL;
+    mEnd   = NULL;
+
+    VerifyOrExit(aUrl != NULL, OT_NOOP);
+    VerifyOrExit(strnlen(aUrl, sizeof(mUrl)) < sizeof(mUrl), OT_NOOP);
+    strncpy(mUrl, aUrl, sizeof(mUrl) - 1);
+    url = strstr(url, "://");
+    VerifyOrExit(url != NULL, OT_NOOP);
+    url += sizeof("://") - 1;
+    mPath = url;
+
+    mStart = strstr(url, "?");
+
+    if (mStart != NULL)
+    {
+        mStart[0] = '\0';
+        mStart++;
+
+        mEnd = mStart + strlen(mStart);
+    }
+    else
+    {
+        mEnd = url + strlen(url);
```

Comment:
From this `else` path, we will have `mStart` as NULL and `mEnd` not NULL.
Later from `GetValue()` we can comapre `start < mEnd` where `start` is set as `mStart` and therefore is `NULL`
IIRC doing `NULL < mEnd` is not well-defined (undefined behavior). Can we consider avoiding this situation? Why do we need `mEnd` when `mStart` is NULL?